src/librustdoc/html/markdown.rs

```diff
@@ -773,7 +794,19 @@ impl<'tcx> ExtraInfo<'tcx> {
         ExtraInfo { def_id, sp, tcx }
     }
 
-    fn error_invalid_codeblock_attr(&self, msg: &str, help: &str) {
+    fn error_invalid_codeblock_attr(&self, msg: &str) {
```

Comment:
This function is called at least once with `&format!(...)` so an allocation is caused and discarded when it could be reused inside instead:

It should probably take `Into<String>` because `struct_span_lint_hir()`'s `msg` parameter is `impl Into<DiagnosticMessage>` and for `&str` that impl is simply `DiagnosticMessage::Str(s.to_string())` -> by using `Into<String>` we could pass a `String` when necessary and avoid the double allocation

A nit, but I like perfs when I can find them :D

Reply:
Or directly `impl Into<DiagnosticMessage>` even, I don't know if that one is accessible in here

Reply:
Very good point!